### PR TITLE
Add code style tools

### PR DESCRIPTION
**Pull request by alex98247**
## Summary

Change pipeline

## Quick changelog

- Add code check style
- Add typing check
- Add precommit hook 
- Separate dev requirements and run requirements

## What's new?

Pipeline will check code style and typing

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+How to prioritize pull requests, from most to least important:
+
+1. Fixes for broken tests. Broken means broken on any supported platform or Python version.
+1. Extra tests to cover corner cases.
+1. Minor edits to docs.
+1. Bug fixes.
+1. Major edits to docs.
+1. Features.
+
```
Comment (f1i3g3) [inline CONTRIBUTING.md:87]:
```
1. Fixes for broken tests. Broken means broken on any supported platform or Python version.
1. Extra tests to cover corner cases.
1. Minor edits to docs.
1. Bug fixes.
1. Major edits to docs.
1. Features.
```
Мне кажется, тут должно быть не так.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+- Create issues for any major changes and enhancements that you wish to make. Discuss things transparently and get community feedback.
+- Keep feature versions as small as possible, preferably one new feature per version.
+- Be welcoming to newcomers and encourage diverse new contributors from all backgrounds. See the Python Community Code of Conduct (https://www.python.org/psf/codeofconduct/).
+
+### Becoming a Committer
+
+Contributors may be given commit privileges. Preference will be given to those with:
+
+1. Past contributions to pysatl-experiment and other related open-source projects. Contributions to pysatl-experiment include both code (both accepted and pending) and friendly participation in the issue tracker and Pull request reviews. Both quantity and quality are considered.
+2. coding style that the other core committers find simple, minimal, and clean.
```
Comment (f1i3g3) [inline CONTRIBUTING.md:121]: `2. coding style that the other core committers find simple, minimal, and clean.`
И вот тут надо будет подправить.

**Review by f1i3g3 — APPROVED** [review]
За исключением парочки опечаток, которые можно потом подправить - все норм.
Ну и еще CI почему-то падает.

**Comment by alex98247** [comment]
> За исключением парочки опечаток, которые можно потом подправить - все норм.
> 
> Ну и еще CI почему-то на импорты ругается.

Там стоит проверка порядка импортов. Я думаю в этом реквесте надо еще коммит сделать с правкой отступов и всего на что ругаются майпай, руф и тд, что б CI был зеленый

Опечатки поправлю
